cmd/clusterctl/client/get_kubeconfig.go

```diff
@@ -56,5 +61,16 @@ func (c *clusterctlClient) GetKubeconfig(options GetKubeconfigOptions) (string,
 		options.Namespace = currentNamespace
 	}
 
-	return clusterClient.WorkloadCluster().GetKubeconfig(options.WorkloadClusterName, options.Namespace)
+	// system kubeconfig is explicitly asked for
+	if !options.UserKubeconfig {
+		return clusterClient.WorkloadCluster().GetKubeconfig(options.WorkloadClusterName, options.Namespace)
+	}
+	kubeconfig, err := clusterClient.WorkloadCluster().GetUserKubeconfig(options.WorkloadClusterName, options.Namespace)
+	if err != nil {
+		// fallback on the system kubeconfig
+		logger := log.Log
+		logger.Info("failed to fetch the user kubeconfig; fetching the system kubeconfig instead")
```

Comment:
Please only return an error here don't log + return an error

+ wrap the error as suggested in the other comments

Reply:
How about doing error check on the call to get the system kubeconfig and log only if are able to get the system kubeconfig?

Reply:
Why do we need it? I would assume:
* no error => we just print the kubeconfig
* error => we print the kubeconfig